### PR TITLE
Warn if a user or user group has permission to import themes

### LINKED ISSUE #853: Verwaltung des Rechts für das Theme-Modul Theme-Import
Wenn einem regulären Benutzer das Recht für das Modul Templates eingeräumt wurde, erscheint im Modul Benutzer eine Warnung: "Mindestens ein regulärer Benutzer hat Zugriff auf den Template-Editor. Da Templates PHP-Dateien sind, haben diese Benutzer implizit die volle Kontrolle über das System!"

1. Wenn einem regulären Benutzer das Recht für das Theme-Modul ``Theme-Import`` eingeräumt wurde, erscheint keine Warnung, obwohl auf diesem Weg ebenfalls der Import von beliebigen Templates möglich ist.

2. Ein regulärer Benutzer kann auch, wenn er aufgrund der Einstellung ``Erlaubte Felder`` in einer seiner Benutzergruppen das Recht für das Feld ``themes`` in ``tl_user`` oder ``tl_user_group`` hat, sich oder anderen Zugriff auf den ``Theme-Import`` verschaffen.
Die Berechtigung für das Theme-Modul ``Theme-Import`` sollte daher nur von Administratoren vergeben werden könnte, unabhängig von der Berechtigung für das Feld ``theme`` in ``tl_user`` oder ``tl_user_group``.

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #853

### CODE REVIEW

File: core-bundle/src/Resources/contao/languages/en/default.xlf [hunk 1]
```diff
@@ -1950,7 +1950,13 @@
         <source>There is at least one regular user with access to the template editor. Since templates are PHP files, these users implicitly have full control over the system!</source>
       </trans-unit>
       <trans-unit id="MSC.groupTemplateEditor">
-        <source>There is at least one user group granting access to the template editor. Since templates are PHP files, users of these groups implicitly have full control over the system!</source>
+        <source>There is at least one user group with access to the template editor. Since templates are PHP files, users of these groups implicitly have full control over the system!</source>
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/languages/en/default.xlf]: I liked the previous text better (*granting* for groups)

**Review by aschempp — DISMISSED** [review]

File: core-bundle/src/Resources/contao/languages/en/default.xlf (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/languages/en/default.xlf]: Changed in 1fa2bb6439a01aff9320f406e950ea5f70efd249.

**Review by aschempp — APPROVED** [review]

**Review by aschempp — CHANGES_REQUESTED** [review]
Can't we count both in the same SQL query though?

**Comment by leofeyer** [comment]
Then we do not know which error message to show, do we?

**Comment by aschempp** [comment]
A quick search gave me the following:

```sql
SELECT 
    SUM(case when themes LIKE '%\"theme_import\"%' then 1 else 0 end) AS themeCount,
    SUM(case when modules LIKE '%\"tpl_editor\"%' then 1 else 0 end) AS tplCount
FROM tl_user
WHERE admin=''
```

**Comment by leofeyer** [comment]
Changed in fc0344623743c45a5fd75d36ccb72d3edeef4eb0.

The new query uses `EXISTS` instead of `SUM`, because as @ausi has explained:

> exists bricht beim ersten match ab... wenn du viele matches hast wird count langsamer, exists nicht... und SUM() hat das selbe problem dass es über alle rows geht...

**Review by Toflar — APPROVED** [review]

**Review by bytehead — APPROVED** [review]

**Review by ausi — APPROVED** [review]

**Review by aschempp — APPROVED** [review]

**Pull request merged**